nessie/src/test/java/org/apache/iceberg/nessie/TestNamespace.java

```diff
@@ -63,14 +65,56 @@ public void testListNamespaces() {
     tables = catalog.listTables(null);
     Assertions.assertThat(tables).isNotNull().hasSize(6);
 
-    List<Namespace> namespaces = catalog.listNamespaces();
-    Assertions.assertThat(namespaces).isNotNull().hasSize(5);
-    namespaces = catalog.listNamespaces(Namespace.of("a"));
-    Assertions.assertThat(namespaces).isNotNull().hasSize(3);
-    namespaces = catalog.listNamespaces(Namespace.of("a", "b"));
-    Assertions.assertThat(namespaces).isNotNull().hasSize(2);
-    namespaces = catalog.listNamespaces(Namespace.of("b"));
-    Assertions.assertThat(namespaces).isNotNull().hasSize(2);
+    List<Namespace> namespaces;
+    if (api instanceof NessieApiV2) {
```

Comment:
Would it be possible to use `@NessieApiVersions` test annotations to avoid this kind of `if` is test code?

Reply:
Whenever test logic different, I think it would be nice to have one method for v1 and another for v2, each annotated with `@NessieApiVersions(...)` for its specific target version. Common code could be refactored into utility methods. WDYT?

Reply:
If there are many differences perhaps we could have a dedicated sub-class for each API version.

Reply:
IMHO, `if`s in the test code complicate understanding of what the test actually covers.